proxy/InkAPIInternal.h

```diff
@@ -265,21 +265,26 @@ class HttpAPIHooks : public FeatureAPIHooks<TSHttpHookID, TS_HTTP_LAST_HOOK>
 {
 };
 
-typedef enum {
-  TS_SSL_INTERNAL_FIRST_HOOK,
-  TS_VCONN_START_INTERNAL_HOOK = TS_SSL_INTERNAL_FIRST_HOOK,
-  TS_VCONN_CLOSE_INTERNAL_HOOK,
-  TS_SSL_CERT_INTERNAL_HOOK,
-  TS_SSL_SERVERNAME_INTERNAL_HOOK,
-  TS_SSL_VERIFY_SERVER_INTERNAL_HOOK,
-  TS_SSL_VERIFY_CLIENT_INTERNAL_HOOK,
-  TS_SSL_SESSION_INTERNAL_HOOK,
-  TS_VCONN_OUTBOUND_START_INTERNAL_HOOK,
-  TS_VCONN_OUTBOUND_CLOSE_INTERNAL_HOOK,
-  TS_SSL_INTERNAL_LAST_HOOK
-} TSSslHookInternalID;
-
-class SslAPIHooks : public FeatureAPIHooks<TSSslHookInternalID, TS_SSL_INTERNAL_LAST_HOOK>
+class TSSslHookInternalID
+{
+public:
+  constexpr TSSslHookInternalID(TSHttpHookID id) : _id(id - TS_SSL_FIRST_HOOK) {}
+
+  constexpr operator int() const { return _id; }
+
+  static const int NUM = TS_SSL_LAST_HOOK - TS_SSL_FIRST_HOOK + 1;
+
+  constexpr bool
+  inBounds() const
```

Comment:
We've been tending more towards STL style method naming, e.g. `is_in_bounds`.

Reply:
I favor the C++PL conventions (with underscores) myself but what about https://cwiki.apache.org/confluence/display/TS/Coding+Style#CodingStyle-Namingconventions

Reply:
Updated.